### PR TITLE
GKCS-5303 Allow the browser extension to connect to On-premise GitHub environments

### PR DESCRIPTION
**Pull request by gitkrakel**
https://gitkraken.atlassian.net/browse/GKCS-5303

Tested with https://ghe.ci.gitkraken.com (running Enterprise v3.11.2).

There are buttons on commit comparison, individual commits, and repo landing pages. (i.e. `< > Code` tab of a repo).

However for PRs, I noticed there's no `< > Code` button, so I didn't add any buttons on PR pages (note: this might be because our instance is behaving weirdly right now: https://gitkraken-hq.slack.com/archives/C05BYV78VQB/p1709170626154339)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -48,11 +48,13 @@
 		"watch:firefox": "yarn run set-manifest:firefox:force && yarn watch"
 	},
 	"dependencies": {
+		"ramda": "^0.29.1",
```
Comment (jdgarcia) [inline package.json:51]: Adding ramda doubles the `background.js` file size. Things like `difference` and `union` might be a bit annoying to write but a lot of the other functions like `map` and `filter` are available natively so I would consider if we actually need this.

With ramda / Without ramda
image

File: src/types.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (jdgarcia) [inline src/types.ts:1]: 👀

**Review by jdgarcia — CHANGES_REQUESTED** [review]
actually, do you mind splitting the enterprise stuff out to a separate PR? the permission stuff by itself is useful and needed, and could be released before the enterprise stuff

File: package.json (hunk 1 above)
Comment (gitkrakel) [reply package.json:51]: Fair point, I think I only really need `difference` so I'll write a util func for that.

**Review by jdgarcia — APPROVED** [review]

**Pull request merged**